### ISSUE TITLE
Monday morning Plover problem

### ISSUE THREAD

**Issue by edeutsch** [RTXteam/RTX#1752]
I first got the alert at 7am and now again at 8am. the KG2 endpoint is failing to return results upon hourly testing with errors such as this:

```
    {
      "code": "",
      "level": "DEBUG",
      "message": "Sending batch 1 to Plover (has 1 input curies)",
      "timestamp": "2021-11-22T16:01:16.565253Z"
    },
    {
      "code": "",
      "level": "WARNING",
      "message": "Plover returned a status code of 502. Response was: <html>\r\n<head><title>502 Bad Gateway</title></head>\r\n<body>\r\n<center><h1>502 Bad Gateway</h1></center>\r\n</body>\r\n</html>\r\n",
      "timestamp": "2021-11-22T16:01:16.855914Z"
    },
    {
      "code": "RequestFailed",
      "level": "ERROR",
      "message": "Plover returned response of 502. Answer was: {}",
      "timestamp": "2021-11-22T16:01:16.856024Z"
    },
```

**Comment by amykglen** [comment]
weird - ok, both KG2 endpoints are working again!

I turned our own Plover instance back on and pointed them to it. guess I spoke too soon about not having issues with the ITRB Plover! I'll message them about it.

**Comment by edeutsch** [comment]
Thank you!!! confirmed working.

**Comment by edeutsch** [comment]
In case it is useful, this system failed its hourly tests at 7am, 8am, 9am. But was fine before that.

**Comment by saramsey** [comment]
TY @edeutsch for logging this issue. Just catching up now... so we think the root cause in this case was a PloverDB outage, is that correct?

**Comment by edeutsch** [comment]
Correct. Specifically the PloverDB instance hosted by ITRB. My understanding is that @amykglen temporarily solved the problem by launching our own PloverDB instance and pointing our KG2 and KG2beta services to this temporary instance.

I'm guessing that a final root cause analysis, return to the previous state, and decision about future plans is still pending.

**Comment by saramsey** [comment]
Thank you @amykglen !

**Comment by amykglen** [comment]
ok, have some info from ITRB now: Kanna said that one of his colleagues was working on a test env and removed the Plover pod. he redployed it and it seems to be working fine now.

do we want to switch back to using the ITRB Plover? I suppose one option could be to use it for now (to save money), but then switch back to our instance in the days leading up to the demo? just a little concerning that the ITRB endpoint went down like this.

**Comment by saramsey** [comment]
> ok, have some info from ITRB now: Kanna said that one of his colleagues was working on a test env and removed the Plover pod. he redployed it and it seems to be working fine now.
> 
> do we want to switch back to using the ITRB Plover? I suppose one option could be to use it for now (to save money), but then switch back to our instance in the days leading up to the demo? just a little concerning that the ITRB endpoint went down like this.

This seems reasonable. For the demo, I think self-hosting in our dedicated EC2 instance makes sense.

**Comment by amykglen** [comment]
ok, cool. I've pointed us back to the ITRB Plover and turned off our Plover instance. verified `kg2` and `kg2beta` seem to be working fine. I'll plan to switch back to our Plover instance in the days leading up to the demo, unless I hear otherwise!
